src/features/ITGovernance/Requester/BusinessCase/AlternativeAnalysis/index.tsx

```diff
@@ -0,0 +1,385 @@
+            return <span className="text-info">In Progress</span>;
+          }
+        }
+      },
+      {
+        // Actions column - shows Edit, Edit and Remove (in case of optional solutions), or blank
+        Header: t<string>('alternativesTable.actions'),
+        Cell: ({ row }: CellProps<ProposedBusinessCaseSolution, string>) => {
+          const { index } = row;
+
+          // Display blank if form hasn't been started yet
+          if (!solutionHasFilledFields(row.original)) {
+            return <></>;
+          }
+
+          // TODO: NJD - super hacky string replace thing, find better way of doing this
+          const newUrl = t(`alternativesTable.solutions.${index}.heading`)
+            .toLowerCase()
+            .replace(/\s+/g, '-')
+            .replace(/alternative-/g, 'alternative-solution-');
+
+          return (
+            <div className="display-flex">
+              <Button
+                type="button"
+                unstyled
+                onClick={() => {
+                  dispatchSave();
+                  history.push(newUrl);
+                }}
+              >
+                {t('general:edit')}
+              </Button>
+              {index > 1 && (
+                <Button
+                  className="margin-left-1"
+                  type="button"
+                  unstyled
+                  // TODO: NJD - need to make this more generic if we start to support more than two alternatives (C, D, etc.)
+                  onClick={async () => {
+                    if (
+                      // eslint-disable-next-line no-alert
+                      window.confirm(t('confirmRemoveAlternativeB'))
+                    ) {
+                      await dispatch(
+                        putBusinessCase({
+                          ...businessCase,
+                          alternativeB: defaultProposedSolution
+                        })
+                      );
+
+                      // Update table data state
+                      setTableData([
+                        businessCase.preferredSolution,
+                        businessCase.alternativeA,
+                        defaultProposedSolution
+                      ]);
+
+                      // Force a re-fetch of the business case data
+
+                      await dispatch(
+                        fetchBusinessCase(businessCase.systemIntakeId)
+                      );
+
+                      // TODO: NJD - this doesn't force a full state refresh. Can call window.location.reload() to fix but feels like bad UX
+                      history.replace(history.location);
+                    }
+                  }}
+                >
+                  {t('general:remove')}
+                </Button>
+              )}
+            </div>
+          );
+        }
+      }
+    ],
+    [businessCase, dispatch, dispatchSave, history, isFinal, t]
+  );
+
+  const table = useTable({
+    columns,
+    data: tableData,
```

Comment:
> In the new AlternativeAnalysis table, I can't get the "remove alternative B" functionality to keep its state without a manual refresh.

It looks like this issue you mentioned in Slack is caused by using the `tableData` state to set the table `data` property. You should be able to remove the `tableData` state altogether and replace `data: tableData` with:
```jsx
data: [
  businessCase.preferredSolution,
  businessCase.alternativeA,
  businessCase.alternativeB
]
```

You would then also remove the `setTableData` code from the code block that defines the `columns`. You can also remove `history.replace(history.location);`. Doing this should result in a table rerender after removing alternative B.